exec/rpc/src/main/java/org/apache/drill/exec/rpc/BasicClient.java

```diff
@@ -69,6 +74,11 @@
   private final IdlePingHandler pingHandler;
   private ConnectionMultiListener.SSLHandshakeListener sslHandshakeListener = null;
 
+  // Authentication related parameters
+  protected volatile List<String> serverAuthMechanisms = null;
```

Comment:
Is `volatile` necessary? Consider making all variables private and providing `setAuthRequired(List<String> authMechanisms)`.